src/skydb.ts

```diff
@@ -1,105 +1,81 @@
-// will try and download the file behind the skylink it has found in the entry.
-export async function getFile(this: SkynetClient, user: User, fileID: FileID): Promise<SkyFile> {
+import { HashRegistryEntry, PublicKey, SecretKey } from "./crypto";
+import { RegistryEntry, SignedRegistryEntry } from "./registry";
+import { parseSkylink, promiseTimeout, trimUriPrefix, uriSkynetPrefix } from "./utils";
+
+export async function getJSON(
+  this: SkynetClient,
+  publicKey: PublicKey,
+  dataKey: string
+): Promise<Record<string, unknown> | null> {
   // lookup the registry entry
-  const existing = await this.lookupRegistry(user, fileID);
-  if (!existing) {
-    throw new Error("not found");
+  const entry: SignedRegistryEntry = await this.registry.getEntry(publicKey, dataKey);
+  if (entry === null) {
+    return null;
   }
 
-  // download the data in that Skylink
-  const skylink = existing.value.data;
+  // Download the data in that Skylink
+  // TODO: Replace with download request method.
+  const skylink = parseSkylink(entry.entry.data);
+
   const response = await this.executeRequest({
     ...this.customOptions,
     method: "get",
     url: this.getSkylinkUrl(skylink),
   });
 
-  // wrap the data in a skyfile and return it
-  const metadata = JSON.parse(response.headers["skynet-file-metadata"]);
-  const file = new SkyFile(new File([response.data], metadata.filename, { type: response.headers["content-type"] }));
-  return file;
+  return response.data;
 }
 
-// setFile uploads a file and sets updates the registry
-export async function setFile(this: SkynetClient, user: User, fileID: FileID, file: SkyFile): Promise<boolean> {
-  // upload the file to acquire its skylink
-  const customFilename = fileID.filename;
-  const skylink = await this.uploadFile(file.file, { customFilename });
-
-  // fetch the current value to find out the revision, timeout after 2s
-  let existing: SignedRegistryValue | null;
-  try {
-    existing = await promiseTimeout(this.lookupRegistry(user, fileID), 2000);
-  } catch (error) {
-    existing = null;
+export async function setJSON(
+  this: SkynetClient,
+  privateKey: SecretKey,
+  dataKey: string,
+  json: Record<string, unknown>,
+  revision?: number
+) {
+  // Upload the data to acquire its skylink
+  // TODO: Replace with upload request method.
+  const file = new File([JSON.stringify(json)], dataKey, { type: "application/json" });
+  const { skylink } = await this.uploadFileRequest(file, this.customOptions);
+
+  const publicKey = pki.ed25519.publicKeyFromPrivateKey({ privateKey });
+  if (!revision) {
+    // fetch the current value to find out the revision.
+    const entry = await this.registry.getEntry(publicKey, dataKey);
+
+    if (entry) {
+      // verify here
+      if (
+        !pki.ed25519.verify({
+          message: HashRegistryEntry(entry.entry),
+          signature: entry.signature,
+          publicKey,
+        })
+      ) {
+        throw new Error("could not verify signature");
```

Comment:
Can we make this error a little more user friendly? Give some more details on what could be wrong.